Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -135,10 +153,21 @@ import BraintreeCore
         context: ASWebAuthenticationPresentationContextProviding,
         completion: @escaping (Bool, Error?) -> Void
     ) {
-        apiClient.sendAnalyticsEvent("ios.sepa-direct-debit.web-flow.started")
-        self.webAuthenticationSession.start(url: url, context: context) { url, error in
-            self.handleWebAuthenticationSessionResult(url: url, error: error, completion: completion)
-        }
+        self.webAuthenticationSession.start(
+            url: url,
+            context: context,
+            sessionDidDisplay: { [weak self] didDisplay in
+                if didDisplay {
+                    self?.apiClient.sendAnalyticsEvent(BTSEPADirectAnalytics.challengePresentationSucceeded)
+                    self?.isAuthenticationSessionStarted = true
+                } else {
+                    self?.apiClient.sendAnalyticsEvent(BTSEPADirectAnalytics.challengePresentationFailed)
+                }
+            },
```

Comment:
We need to add a unit test for this added logic. That said, all of the analytic event audit PRs need follow up Unit Test PRs. There are several events that are sent, but never validated for when they should be sent in a unit test.

Victoria - this is something we should make tickets for.

Reply:
@KunJeongPark - can you link the ticket you made for unit testing this?

Reply:
https://paypal.atlassian.net/browse/DTBTSDK-2653 => I made one for each module that we did analytics audit for. This one is for SEPA.

Reply:
Awesome, thank you!